src/editor/model.ts

```diff
@@ -65,6 +65,43 @@ interface ITextChange {
 
 
 /**
+ * An interface describing editor text changes.
+ */
+export
+interface ICompletionRequest {
```

Comment:
Why not unify this with `ITextChange` into an `IEditorState`, that could also be exposed as a `getState()` method on the Editor widget?